sfge/src/main/java/com/salesforce/cli/CliArgParser.java

```diff
@@ -41,6 +41,48 @@ public CLI_ACTION getCliAction(String... args) {
         }
     }
 
+    public static class CatalogArgParser {
+        private static final int ARG_COUNT = 2;
+        // NOTE: This value must match the one for the RuleType enum declared in Constants.ts.
+        private static final String PATHLESS = "pathless";
+        // NOTE: This value must match the one for the RuleType enum declared in Constants.ts.
+        private static final String DFA = "dfa";
```

Comment:
For the names, I've been using the terms "AST-based", "Graph-based", and "Path-based" to show the different levels of analysis we can perform from Code Analyzer. The idea is that each level builds an additional capability on the previous level. How about the terms "graph-based" and "path-based"?

Reply:
That's definitely an option. I'll see where `Pathless` is used in the codebase and how easy/sensible it would be to change that.